crates/ruff/src/pyproject_toml.rs

```diff
@@ -0,0 +1,36 @@
+use anyhow::Result;
+use pyproject_toml::PyProjectToml;
+use ruff_text_size::{TextRange, TextSize};
+
+use ruff_diagnostics::Diagnostic;
+use ruff_python_ast::source_code::SourceFile;
+
+use crate::message::Message;
+use crate::rules::ruff::rules::InvalidPyprojectToml;
+
+pub fn lint_pyproject_toml(source_file: SourceFile) -> Result<Vec<Message>> {
+    let err = match toml::from_str::<PyProjectToml>(source_file.source_text()) {
+        Ok(_) => return Ok(Vec::default()),
+        Err(err) => err,
+    };
+
+    let range = match err.span() {
+        // This is bad but sometimes toml and/or serde just don't give us spans
+        None => TextRange::default(),
+        Some(range) => {
+            let expect_err = "pyproject.toml file be smaller than 4GB";
```

Comment:
```suggestion
            let expect_err = "pyproject.toml file to be smaller than 4GB";
```

Or we could even output this as a diagnostic or handle it generically inside of the CLI (calling the RustPython parser with a string longer than 4GB panics right now)

Reply:
Changed this to emitting a diagnostic, no reason to panic